src/secret_manager/infisical.ts

```diff
@@ -0,0 +1,111 @@
+import { InfisicalSDK } from '@infisical/sdk'
+import { AbstractSecretManager } from "./abstractSecretManager";
+
+export class Infisical extends AbstractSecretManager {
+
+    clientId: string;
+    clientSecret: string;
+    projectId: string;
+    client: InfisicalSDK;
+    env: string;
+
+    constructor() {
+        if (!process.env.SECRET_MANAGER_INFISICAL_API_URI) {
+            throw new Error("SECRET_MANAGER_INFISICAL_API_URI environment variable is required");
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_CLIENT_ID) {
+            throw new Error("SECRET_MANAGER_INFISICAL_CLIENT_ID environment variable is required");
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_CLIENT_SECRET) {
+            throw new Error("SECRET_MANAGER_INFISICAL_CLIENT_SECRET environment variable is required");
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_PROJECT_ID) {
+            throw new Error("SECRET_MANAGER_INFISICAL_PROJECT_ID environment variable is required");
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_ENV) {
+            throw new Error("SECRET_MANAGER_INFISICAL_ENV environment variable is required");
+        }
+
+        super();
+        this.clientId = process.env.SECRET_MANAGER_INFISICAL_CLIENT_ID || "";
```

Comment:
Merge master in your branch, then use `utils.getEnvVar`.
See https://github.com/invoice-collector/invoice-collector/blob/208ba8665430c5469b43361a2291e787f5b0865c/src/secret_manager/bitwarden.ts#L16C28-L16C43

Remove the if conditions in constructor as well.